app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningFragment.kt

```diff
@@ -60,13 +57,58 @@ class LearningFragment : Fragment() {
         datasetId = args.datasetId
         lifecycleScope.launch {
             dataset = dbMgt.getDatasetById(datasetId)!!
-            learningPresenter.learningMode = args.learningMode
-            learningPresenter.dataset = dataset
-            initLearningViews()
+            targetImageViews = adaptDisplayToNumberOfCategories(dataset)
+
+            learningPresenter = LearningPresenter(dbMgt, args.learningMode, dataset)
+            learningPresenter.updateForNextSorting(
+                parentActivity,
+                targetImageViews,
+                binding.learningToSort
+            )
+            // sets the listeners for the image views of the sorting
+            setEventListeners()
         }
         requireActivity().onBackPressedDispatcher.addCallback(callback)
     }
 
+    /**
+     * This function adapts the display to the number of categories in the dataset
+     * For example, if the dataset only has 2 categories, one of the categories will not be displayed
+     * @param dataset dataset that is used for the learning
+     * @return the ImageViews that stay displayed on screen
+     */
+    private fun adaptDisplayToNumberOfCategories(dataset: Dataset): List<ImageView> {
+        val nbCategories = dataset.categories.size
+        require(nbCategories > 0) {
+            "A dataset used for learning should have at least one category"
+        }
+        val makeInvisible = { v: List<View> -> v.forEach { it.visibility = View.INVISIBLE } }
+        return with(binding) {
+            when (nbCategories) {
+                1 -> {
+                    makeInvisible(listOf(learningCat0, learningCat2))
+                    listOf(learningCat1)
+                }
+                2 -> {
+                    makeInvisible(listOf(learningCat0))
+                    listOf(learningCat1, learningCat2)
+                }
+                else -> listOf(learningCat0, learningCat1, learningCat2)
+            }
+        }
+    }
+
+    /**
+     * TODO
+     */
```

Comment:
This TODO (and others similar to this one) is about the KDoc not the content of the method right?

Reply:
Yes, thanks for reminding me, I'll add it now